boards: x86: add support for UP Squared (Pentium/Celeron)

dcpleung opened the pull request:
This patch set contains the SoC configuration for Apollo Lake, and the board configuration for UP Squared board. This allows Zephyr to run on the UP Squared board.

dcpleung committed:
boards: x86: add support for UP Squared (Pentium/Celeron)

This adds a primitive board configuration for the UP Squared board
containing Apollo Lake based Pentinum and Celeron SoC. This has
been tested on model UPS-APLP4-A10-0432.

This starts from the minnowboard configuration, and document
from galileo.

Origin: Original

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+
+.. note::
+   This board configuration only works with the boards containing
+   Intel |reg| Pentium |trade| SoC, or Intel |reg| Celeton |trade| SoC.
```
``Celeron`` (misspelled)

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+|           |            | serial port-interrupt |
++-----------+------------+-----------------------+
+
+The kernel currently does not support other hardware features.
```
To clarify,

    The Zephyr kernel currently ...

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+HPET System Clock Support
+-------------------------
+
+Galileo uses HPET timing with legacy-free timer support. The board
```
Replace ``Galileo`` with ``UP Squared``  :)

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+Programming and Debugging
+*************************
+
+Use the following procedures for booting an image on a Galileo board.
```
Replace ``Galileo`` with ``UP Squared``  :)

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+a UP Squared board.
+
+#. Build a Zephyr application; for instance, to build the ``hello_world``
+   application on Galileo:
```
Replace ``Galileo`` with ``UP Squared``  :)

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+      created in the build directory after the application is built. This image
+      has removed debug information from the :file:`zephyr.elf` file.
+
+#. Refer to the UP Squared Serial Console Wiki page for instructions on how to
```
Get the link working with:

    #. Refer to the `UP Squared Serial Console Wiki page
       <https://wiki.up-community.org/Serial_console>`_ for instructions on how to
       connect to a serial console.

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+
+   `<https://wiki.up-community.org/Serial_console>`_
+
+#. Format the USB flash drive as FAT
```
I'd put in ``code-block``s with the actual commands to use for these steps.

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+
+2. Connect the board to the host system using the serial cable and
+   configure your host system to watch for serial data.  See
+   `https://wiki.up-community.org/Serial_console`_.
```
Replace this line with just the url:

    https://wiki.up-community.org/Serial_console

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+   Zephyr application image.
+
+   .. note::
+      If the following messages appear during boot, they can be safely
```
change to:

    You can safely ignore this message if it appears:

dbkinder commented on boards/x86/up_squared/doc/up_squared.rst:
```diff
@@ -0,0 +1,238 @@
+
+         WARNING: no console will be available to OS
+
+References
```
Delete this title (and underline)

dcpleung commented:
Updated the documentation for comments from @dbkinder .

ramakrishnapallala commented on arch/x86/soc/apollo_lake/Kconfig.defconfig:
```diff
@@ -0,0 +1,49 @@
+#
+# Kconfig - Apollo Lake SoC configuration options
+#
+# Copyright (c) 2014-2015 Wind River Systems, Inc.
```
this is new file anyway, why do we need this copyright?

dcpleung replied:
The file was copied from the arch/x86/soc/atom. So need to preserve copyright.

ramakrishnapallala commented on arch/x86/soc/apollo_lake/linker.ld:
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2011-2014, Wind River Systems, Inc.
```
Intel 2018?

dcpleung replied:
The file was copied from the arch/x86/soc/atom. Nothing changed so preserve the original copyright.

ramakrishnapallala commented on arch/x86/soc/apollo_lake/soc.h:
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2010-2015, Wind River Systems, Inc.
```
why this line is needed?

dcpleung replied:
The file was copied from the arch/x86/soc/atom. So need to preserve copyright.

nashif replied:
You should put your copyright on top if you have made any changes

ramakrishnapallala commented on boards/x86/up_squared/up_squared.yaml:
```diff
@@ -0,0 +1,11 @@
+identifier: up_squared
+name: UP Squared
+type: mcu
```
is mcu is the right name for Apollo lake?

dcpleung replied:
scripts/sanity_chk/sanitycheck-platform-schema.yaml does not seem to have any other suitable type. Do you want to add one?

ramakrishnapallala commented on boards/x86/up_squared/up_squared_defconfig:
```diff
@@ -0,0 +1,15 @@
+CONFIG_X86=y
+CONFIG_SOC_APOLLO_LAKE=y
+CONFIG_BOARD_UP_SQUARED=y
+CONFIG_CPU_APOLLO_LAKE=y
```
what is the difference b/w CONFIG_CPU_APOLLO_LAKE & CONFIG_SOC_APOLLO_LAKE? i mean can we have any other configuration with CONFIG_CPU_APOLLO_LAKE other than CONFIG_SOC_APOLLO_LAKE?

dcpleung replied:
The SOC will select the CPU, so will remove the CPU_* here.

ramakrishnapallala reviewed (commented):
minor comments and queries.

dcpleung commented on arch/x86/soc/apollo_lake/Kconfig.soc:
```diff
@@ -0,0 +1,14 @@
+	select CPU_APOLLO_LAKE
+	select BOOTLOADER_UNKNOWN
+	select X86_MMU
+	select APPLICATION_MEMORY
```
These were copied from the Atom SoC Kconfig. Will remove.

dcpleung commented:
Removed the lines as indicated by the reviews.

dcpleung committed:
arch/x86/soc: add SoC configuration for Apollo Lake

This adds the SoC configuration for Apollo Lake. This is based
on the Atom configuration.

Origin: Original

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
boards/x86: scripts: extend build_grub.sh for 64-bit UEFI

This extends the build_grub.sh script so that it can build 64-bit
Grub UEFI application, and thus enabling us to boot on systems
with 64-bit BIOS.

Also updated the script to checkout the tag grub-2.02 since
v2.02 has been out.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung commented:
Moved the copyright lines.
